mitodl/open-discussions
# Cc/url widget

#### Pre-Flight checklist



#### What are the relevant tickets?
#3695 

#### What's this PR do?
This field enables the Embedded URL widget for infinite corridor

#### How should this be manually tested?
In a field that you have moderator rights for (either from being superuser or explicit permissions):
1. Navigate to field page, click gear icon, click "Manage widgets"
2. Add a new widget of type Embedded URL
3. Paste a URL into the field. Try youtube videos and other sites. The preview is powered by [embedly cards](https://embed.ly/cards), just like in Open Discussions.

    _Note: Not at all sites have great embedly previews... I need to learn more about this._
4. Try editing / saving / viewing
5. Try invalid urls

Known Issues:
- Drag reordering a card with an embedly preview temporarily breaks the preview... It can be fixed by collapsing and expanding the widget. Open Discussions has the same issue. I did not look into it too much.
- The cards get too narrow on Mobile screens. I would like to address this separately and will make a separate issue for it.
- custom_html intentionally does not work right now. We will remove it. See  https://github.com/mitodl/hq/discussions/239

#### Screenshots (if appropriate)
<img width="1502" alt="Screen Shot 2022-09-06 at 7 00 41 AM" src="https://user-images.githubusercontent.com/9010790/188622470-750e18ce-9ed7-4c04-8e42-7af9e1c945d9.png">

<img width="931" alt="Screen Shot 2022-09-06 at 7 14 05 AM" src="https://user-images.githubusercontent.com/9010790/188622503-53475322-d679-4c79-8636-f6196b238b31.png">
